### PR TITLE
fixed optional loading functionality & get test cases to pass

### PR DESCRIPTION
**Pull request by zeryx**
the optionality of load was bugged as we never checked for the default case of the function, this yielded a dead load_func which failed in getfuncargs

Also, we added github actions and we had some issues with version compatibility, this fixes those issues as well.

### CODE REVIEW

**Comment by zeryx** [comment]
python 2.7 still fails, will be solved in a separate release.

**Comment by zeryx** [comment]
made changes to make multi-version compatible

File: Test/algo_test.py [hunk 1]
```diff
@@ -29,18 +29,18 @@ def test_text_unicode(self):
...
-        #Unicode input to pipe()
-        result1 = self.client.algo('util/Echo').pipe(telephone)
-        self.assertEquals('json', result1.metadata.content_type)
-        self.assertEquals(telephone, result1.result)
-
-        #Unicode return in .result
-        result2 = self.client.algo('util/Echo').pipe(result1.result)
-        self.assertEquals('json', result2.metadata.content_type)
-        self.assertEquals(telephone, result2.result)
+    # def test_json_unicode(self):
```
Comment (zeryx) [inline Test/algo_test.py]: @pmcq @jamesatha if you guys have some insights into this error case that would be great - it doesn't seem to succeed at all anymore and I'm unsure as to what changed.

Comment (pmcq) [reply Test/algo_test.py]: i tried running these locally and it seems like the client isn't sending the initial request as JSON any longer but as a text string, i'm not sure if these tests just haven't been run for awhile or what though.

Comment (zeryx) [reply Test/algo_test.py]: yea that's what I was thinking - I don't see anything specific that it's testing but I also don't want to remove a critical test that we've just forgotten about over the years.

**Comment by zeryx** [comment]
Closing this by EoD

**Comment by zeryx** [comment]
> seems okay - can you at least file a ticket to fix up the test?

already ahead of you: https://github.com/algorithmiaio/algorithmia-python/issues/25

**Pull request merged**